### PR TITLE
Ceilometer events support

### PR DESCRIPTION
**Pull request by leifmadsen**
This pull request replaces #71

### CODE REVIEW

File: internal/pkg/events/incoming/ceilometer.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		"AUDIT":    "info",
+		"INFO":     "info",
+		"WARN":     "warning",
+		"WARNING":  "warning",
+		"CRITICAL": "critical",
+		"ERROR":    "critical",
+	}
+)
+
+type AlertKeySurrogate struct {
```
Comment (pkilambi) [inline internal/pkg/events/incoming/ceilometer.go]: Can you add a comment to this type to make go lint happy?

Comment (leifmadsen) [reply internal/pkg/events/incoming/ceilometer.go]: That's next on my list. Just gotta get the kids and do a meeting first.

**Comment by leifmadsen** [comment]
# Testing

* enabled `debug: true` on ceilometer SG
* updated address to point at `anycast/ceilometer/events.sample` to get data
* triggered an event from the OpenStack side

Result: FAILED

Document isn't written to ElasticSearch as it is rejected with error 400.

```
- error: elastic: Error 400 (Bad Request): mapper [payload.traits] of different type, current_type [long], merged_type [text] [type=illegal_argument_exception]
- event: &{{"message_id": "a149fd61-1d2f-45d3-a43c-c780a6d17b35", "publisher_id": "telemetry.publisher.controller-0.redhat.local", "event_type": "event", "priority": "SAMPLE", "payload": [{"message_id": "0197fa42-74ab-4d83-88d1-ac2779a34e4e", "event_type": "image.update", "generated": "2020-03-06T14:23:28.675514", "traits": [["service", 1, "image.localhost"], ["project_id", 1, "0f500647077b47f08a8ca9181e9b7aef"], ["user_id", 1, "0f500647077b47f08a8ca9181e9b7aef"], ["resource_id", 1, "07f26e56-ca78-4fa7-bda5-7314b55d94fc"], ["name", 1, "cirros"], ["status", 1, "active"], ["created_at", 4, "2020-03-06T14:23:27"], ["size", 2, 13287936]], "raw": {}, "message_signature": "a5cc0113da46f77499e66adcadf351ba2e178e4c849290d6d5d9abfe18b93cc4"}], "timestamp": "2020-03-06 14:23:28.683488"} map[event_type:event message_id:a149fd61-1d2f-45d3-a43c-c780a6d17b35 payload:[map[event_type:image.update generated:2020-03-06T14:23:28.675514 message_id:0197fa42-74ab-4d83-88d1-ac2779a34e4e message_signature:a5cc0113da46f77499e66adcadf351ba2e178e4c849290d6d5d9abfe18b93cc4 raw:map[] traits:[[service %!s(float64=1) image.localhost] [project_id %!s(float64=1) 0f500647077b47f08a8ca9181e9b7aef] [user_id %!s(float64=1) 0f500647077b47f08a8ca9181e9b7aef] [resource_id %!s(float64=1) 07f26e56-ca78-4fa7-bda5-7314b55d94fc] [name %!s(float64=1) cirros] [status %!s(float64=1) active] [created_at %!s(float64=4) 2020-03-06T14:23:27] [size %!s(float64=2) %!s(float64=1.3287936e+07)]]]] priority:SAMPLE publisher_id:telemetry.publisher.controller-0.redhat.local timestamp:2020-03-06 14:23:28.683488] ceilometer_}
```

File: internal/pkg/events/incoming/ceilometer.go [hunk 2]
```diff
@@ -98,6 +105,45 @@ func (evt *CeilometerEvent) ParseEvent(data string) error {
 		log.Fatal(err)
 		return err
 	}
+	// transforms traits key into map[string]interface{}
```
Comment (pleimer) [inline internal/pkg/events/incoming/ceilometer.go]: Storage to ES currently uses the sanitized json string, so I do not think these changes are even being used: https://github.com/infrawatch/smart-gateway/blob/80caab3837a3876381947cdfc8a141b6543b57f7/internal/pkg/events/events.go#L321

The ElasticClient Create() function will also need to be modified from using BodyString() to BodyJson(): https://github.com/infrawatch/smart-gateway/blob/80caab3837a3876381947cdfc8a141b6543b57f7/internal/pkg/saelastic/client.go#L149

Comment (pleimer) [reply internal/pkg/events/incoming/ceilometer.go]: I took the liberty of making those changes, and it passes!

```
=== RUN   TestStoreCeilometerEvent
=== RUN   TestStoreCeilometerEvent/Store_event_with_differing_trait_types
newPayload: map[event_type:image.delete generated:2020-03-06T14:13:29.497096 message_id:084c0bca-0d19-40c0-a724-9916e4815845 message_signature:77e798b842991f9c0c35bda265fdf86075b4a1e58309db1d2adbf89386a3859e raw:map[] traits:map[created_at:2020-03-06T14:01:07 deleted_at:2020-03-06T14:13:29 name:cirros project_id:0f500647077b47f08a8ca9181e9b7aef resource_id:c4f7e00b-df85-4b77-9e1a-26a1de4d5735 service:image.localhost size:1.3287936e+07 status:deleted user_id:0f500647077b47f08a8ca9181e9b7aef]]
--- PASS: TestStoreCeilometerEvent (0.03s)
    --- PASS: TestStoreCeilometerEvent/Store_event_with_differing_trait_types (0.03s)
```

**Comment by leifmadsen** [comment]
Talked with @pleimer and I'm good with this merging as is assuming the initial scenario is working. We can iterate and fix additional scenarios and such going forward.

**Comment by leifmadsen** [comment]
Paul did another test run and confirms it works. Good enough for me! Merging.

**Pull request merged**